### ISSUE TITLE
Stack Trace on watching any YouTube linked video

### ISSUE THREAD

**Issue by JeremyNichvalodoff** [bdecentgmbh/moodle-mod_videotime#47]
Hi,

We are getting an error on all the VideoTime activities we setup. Says's invalid record then shows the error thats attached.

![image001](https://github.com/bdecentgmbh/moodle-mod_videotime/assets/66082013/0af95f44-453c-44af-ae17-f668fcdfc40f)

**Comment by stefanscholz** [comment]
Hi @JeremyNichvalodoff ,

thanks for reporting this issue.
Could you provide a litte more context? Specifically, which version of Moodle and Video Time you are using would be important.

Video Time 1.7 has two versions at the moment: one for 4.2 and one for 4.1 — the stack trace mentions external functions, which was the area most affected by the change from 4.1 to 4.2 and required us to create a separate version for it.

Best
Stefan

**Comment by JeremyNichvalodoff** [comment]
Hi Stefan,

We click into the course then we see the Video on the course as we set it to label view
<img width="1348" alt="Screenshot 2023-06-12 at 8 55 51 AM" src="https://github.com/bdecentgmbh/moodle-mod_videotime/assets/66082013/8ba0557f-002a-4685-8cae-c032ae49ce2f">
Then we click on the video to start it
<img width="1348" alt="Screenshot 2023-06-12 at 8 56 09 AM" src="https://github.com/bdecentgmbh/moodle-mod_videotime/assets/66082013/3956cac9-cea4-4fd7-a9de-d81e1aec3dc0">


Moodle 4.1.3 +



Video Time Pro videotimeplugin_pro | 1.7 2023011201
Video Time VideoJS Player videotimeplugin_videojs | 1.7 2023011200
Video Time Vimeo Player videotimeplugin_vimeo | 1.7 2023011200

**Comment by stefanscholz** [comment]
Hi @JeremyNichvalodoff,

thanks for the prompt response. 

How did you install Video Time Pro? Did you use the installation file provided which included the pro version? If yes, then the versions should match. Looking at the version numbers, your pro is slightly older than the most recent one (latest is 2023011202). The most recent release is a bugfix release, but your error does not ring any bells to any issue we had before.

Is it happening for any youtube video? Or just the ones from your own account? Does it for example work with this video https://youtube.com/watch?v=qPpCmuAGgtM?

Best
Stefan

**Comment by stefanscholz** [comment]
If your platform is public, would it be possible to have a look at it? You can send the credentials in private to our support channel (support@bdecent.de).

**Comment by JeremyNichvalodoff** [comment]
<img width="1348" alt="Screenshot 2023-06-12 at 9 43 26 AM" src="https://github.com/bdecentgmbh/moodle-mod_videotime/assets/66082013/625ce628-0bf1-488f-8aa7-236071ea784c">
Happens on that link as well.

**Comment by JeremyNichvalodoff** [comment]
Yes, our customer downloaded the installation files from their dashboard for their company.

**Comment by stefanscholz** [comment]
What database are you using? We test against psql and mariadb.

**Comment by SamMcCullough** [comment]
Hi @stefanscholz,

I am working with Jeremy with our customer reporting this issue. We are using AWS Aurora PostgreSQL to host the database. AWS uses PSQL 13.10. 

We are looking to install the latest version of the plugins into their Moodle 4.1.3+ site. We have the plugin package containing  the 1.7.0-r5 mod_videotime with pro plugins included from our customer. They also provided us with the 1.7.0-r3 atto, filter, mod, and pro plugins.

**Comment by SamMcCullough** [comment]
Hi @stefanscholz,

The upgrade from 1.7 to 1.7.0-r5 has resolved the issue for us. I appreciate the support today.

**Comment by stefanscholz** [comment]
Hi @SamMcCullough ,

thanks for letting me know. I suppose the issue was caused by non-matching versions of Video Time and Video Time Pro. That's the number one reason for issues. We're trying to prevent that by shipping one installation file now (instead of several like we did in the past), but it's still possible for example if the module alone is updated, but not the pro version, and we don't really have many options to prevent that. 

Anyway, glad it works now. Closing this issue.

Best
Stefan
